src/components/wallet/methods/updateAccount.ts

```diff
@@ -30,13 +31,37 @@ export default async function updateAccount(this: Wallet) {
       state: loOmit(account, ['id', '_links', 'account_id', 'paging_token']),
       claimableBalances: claimableBalanceResp.records,
     }
+    const balKeystore = await get('BALANCE[keystore]')
+    // this restores the balance prop from storage
+    if (balKeystore && !this.balance.size) {
+      this.balance = new Map()
+      JSON.parse(atob(balKeystore)).forEach((b) => {
+        this.balance.set(b[0], b[1])
+      })
+    }
     account.balances.forEach((b) => {
```

Comment:
I think you need to `set('BALANCE[keystore]', ...)` if there are any new assets in the account right?

What I would do is merge `account.balances` with `this.balance` so that `this.balance` is up to date with whats in the Stellar account PLUS the balances that don't have a trustline.